examples/agentInteractionsDemo/app.js

```diff
@@ -194,53 +269,12 @@ new Vue({
-                const startTime = moment('1997-01-01').toISOString();
-                const endTime = moment().toISOString();
-
-                getEvaluations(qualityApi, startTime, endTime, agentUserId)
-                    .then((data) => {
-                        // Process evaluations data
-                        const evaluations = data.entities;
-                        evaluations.forEach((eval) => {
-                            conversationsApi.getConversation(eval.conversation.id)
-                                .then((conv) => {
-                                    let newConv = conv;
-                                    newConv.customer = conv.participants.find((part) => {
-                                        return part.purpose === "customer";
-                                    });
-                                    this.conversationsData.conversations.push(newConv);
-                                    this.conversationsData.conversations = _.uniqWith(this.conversationsData.conversations, _.isEqual);
-                                    if (this.conversationsData.convEvalMap.has(conv.id)) {
-                                        this.conversationsData.convEvalMap.get(conv.id).push(eval);
-                                    } else {
-                                        this.conversationsData.convEvalMap.set(conv.id, [eval]);
-                                    }
-                                })
-                                .catch((err) => {
-                                    console.log(`Error: ${err}`);
-                                });
-                        });
-                    })
-                    .catch((err) => {
-                        console.log(`Error: ${err}`);
-                    });
-
-                conversationsApi.getConversations()
-                    .then((data) => {
-                        data.entities.forEach((conv) => {
-                            let newConv = conv;
-                            newConv.customer = conv.participants.find((part) => {
-                                return part.purpose === "customer";
-                            });
-                            this.conversationsData.conversations.push(newConv);
-
-                        });
-                        this.conversationsData.conversations = _.uniqWith(this.conversationsData.conversations, _.isEqual);
-                    })
-                    .catch((err) => {
-                        console.log(`Error: ${err}`);
-                    });
+                getConversationsAndEvaluations(agentUserId).then(conversations => {
+                    for(var convId in conversations){
+                        this.conversationsData.conversations.push(conversations[convId].conv)
+                        if(conversations[convId].evals.length > 0) this.conversationsData.convEvalMap.set(convId, conversations[convId].evals);
+                    }
+                }) 
```

Comment:
might be worth adding a catch block here and setting `this.errorMessage` to `Failed to fectch conversations/evaluations`.